core/lib/spree/service_module.rb

```diff
@@ -0,0 +1,64 @@
+module Spree
+  module ServiceModule
+    class MethodNotImplemented < StandardError; end
+    class WrongDataPassed < StandardError; end
+    class NonCallablePassedToRun < StandardError; end
+    class CallMethodNotImplemented < StandardError; end
+
+    Result = Struct.new(:success, :value) do
+      def success?
+        success
+      end
+
+      def failure?
+        !success
+      end
+    end
+
+    module Base
+      def call(input = nil)
+        input ||= {}
```

Comment:
What about `def call(input = {})` ?

Reply:
It's kind of fail-safe if someone would pass `nil` as param by accident. This form was introduced after I did mutation tests with https://github.com/mbj/mutant